helm-charts/splunk-otel-collector/templates/config/_common.tpl

```diff
@@ -209,6 +209,34 @@ k8sattributes:
       {{- include "splunk-otel-collector.addExtraLabels" . | nindent 6 }}
     {{- end }}
 {{- end }}
+{{- define "splunk-otel-collector.k8sClusterReceiverAttributesProcessor" -}}
+k8sattributes/clusterReceiver:
```

Comment:
If we're adding this change, we need to address some cleanup to prevent code duplication related to setting Kubernetes attributes:
- Remove redundant attribute setting: We should clean up any attributes set using the previous method that will now be handled by the k8sattributes processor.
  - [Reference line in PR](https://github.com/signalfx/splunk-otel-collector-chart/pull/1481/files#diff-069710df8ca3984808067088bb09303468c24d7e97e4d8970234e703ef52ffd2R121)
- Clean up unused environment variables: The environment variables in the cluster receiver deployment that are no longer used to set attributes the old way should also be removed.
  - [Env var reference 1](https://github.com/signalfx/splunk-otel-collector-chart/blob/e394d52af4f7a065952737a3bcc9bb0b6b94757b/helm-charts/splunk-otel-collector/templates/deployment-cluster-receiver.yaml#L94)
  - [Env var reference 2](https://github.com/signalfx/splunk-otel-collector-chart/blob/main/helm-charts/splunk-otel-collector/templates/deployment-cluster-receiver.yaml#L132)

Reply:
I removed variables - changes are on separate branch: [here](https://github.com/pszkamruk-splunk/splunk-otel-collector-chart/commit/556fad616e152c214c1fb7f7ff2604ba7752e3fd)
I checked that, after removing env variables from configuration, attributes are still being added to events using k8sAttributesProcessor.
There is another [PR](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35528) in contrib repo which fixes some logs which are not matched by k8sAttributesProcessor. 
In case of these missing/not matched events, env variables are causing that these events have also additional attributes like k8s.pod.name, etc

I noticed also that these env variables are also used to add attributes to metrics by `resource/add_collector_k8s`: [link](https://github.com/signalfx/splunk-otel-collector-chart/blob/7ea0cab65fd0825b61e162e405ae11bedcf06890/helm-charts/splunk-otel-collector/templates/config/_otel-k8s-cluster-receiver-config.tpl#L121) 
There are 3 metric pipelines and one of them (`metrics/collector` pipeline) is using these env variables so I am not sure if we can remove these variables.
But on the other hand all metric pipelines are also using k8sAttributesProcessor, and this also duplicates the functionality of adding attributes to metric pipeline. 

So regarding all above, my questions:
1. If the changes on separate branch are OK, I will merge them here 
2. Should I also remove `resource/add_collector_k8s` which is used by metric pipeline?

Reply:
Let's move this change of removing environment variables to separate PR.
Before removing environment variables, we first need to merge contrinb [PR](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35528) - adding namespace attribute to k8s events allowing k8sAttributesProcessor to match all k8s events. Currently K8sAttributesProcessor does not matching all k8s events.
We need also decide what to do with  `resource/add_collector_k8s`: [link](https://github.com/signalfx/splunk-otel-collector-chart/blob/7ea0cab65fd0825b61e162e405ae11bedcf06890/helm-charts/splunk-otel-collector/templates/config/_otel-k8s-cluster-receiver-config.tpl#L121) 
as this is using env variables which we want to remove, but the functionality of adding additional parameters is covered by `k8sattributes/metrics`, so probably it is safe to remove `resource/add_collector_k8s` from configuration

Reply:
- Environment Variables
  - Remove the environment variables in a separate PR after the contrib fix is merged.
- resource/add_collector_k8s:
  - Replace this with a k8sattributes processor.
  - Aim to use a single instance of the k8sattributes processor if possible. If required, add a second instance (e.g., k8sattributes/add_collector) for handling collector metrics specifically.
  - Functional Tests cover internal collector metrics and attributes (e.g., [otelcol_exporter_send_failed_spans](https://github.com/signalfx/splunk-otel-collector-chart/blob/0e083695672ef3055a008964f6c33675dea236f4/functional_tests/testdata/expected_kind_values/expected_internal_metrics.yaml#L5044)). Check to see if you can get the tests to pass with using only a single k8sattributes processor instance. Add a second processor instance (named something like `k8sattributes/add_collector`) because the tests failed and it is necessary.